### PR TITLE
What If... our Go dependencies were current?

### LINKED ISSUE #381: Please consider dropping support for older golang versions like go1.15 and go1.16
https://github.com/segmentio/chamber/blob/a1f0b44bbd39de50d77702fa7428d613448d7d6d/.github/workflows/build.yml#L14

https://github.com/segmentio/chamber/blob/a1f0b44bbd39de50d77702fa7428d613448d7d6d/go.mod#L21

### PR DESCRIPTION
**Pull request by mckern**
What-if-Watcher

This is ground work for some other maintenance PRs that are coming over the wire soon. All existing tests are passing for this, but test coverage is still kind-of low so take this with a grain of salt.

```
$ go test -cover ./...
?   	github.com/segmentio/chamber/v2	[no test files]
ok  	github.com/segmentio/chamber/v2/cmd	1.001s	coverage: 18.1% of statements
ok  	github.com/segmentio/chamber/v2/environ	0.633s	coverage: 58.0% of statements
ok  	github.com/segmentio/chamber/v2/store	9.671s	coverage: 46.9% of statements
ok  	github.com/segmentio/chamber/v2/utils	0.940s	coverage: 100.0% of statements
```

closes #381

### CODE REVIEW

**Review by nitrocode — DISMISSED** [review]

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -2,33 +2,33 @@ name: Release
...
-      - 'v[0-9]+.[0-9]+.[0-9]-[a-zA-Z0-9]+'
+      - "v[0-9]+.[0-9]+.[0-9]+"
+      - "v[0-9]+.[0-9]+.[0-9]-[a-zA-Z0-9]+"
 
 jobs:
   dist:
     strategy:
       matrix:
-        go: ['1.19.x']
+        go: ["1.19.x"]
```
Comment (alecjacobs5401) [inline .github/workflows/release.yml:12]: given that our `Dockerfile` is now `1.20.4`, do we want to swap this release strategy to be `1.20.x`?

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -37,54 +37,53 @@ jobs:
...
+          generate_release_notes: true
+          fail_on_unmatched_files: true
+          files: |
+            dist/*
 
   publish-dockerhub:
     strategy:
       matrix:
-        go: ['1.19.x']
+        go: ["1.19.x"]
```
Comment (alecjacobs5401) [inline .github/workflows/release.yml:59]: same question applies here

**Review by alecjacobs5401 — DISMISSED** [review]
Two minor comments that I don't think are necessarily blockers for this PR specifically.

Thanks for doing this Ryan!

File: .github/workflows/release.yml (hunk 2 above)
Comment (mckern) [reply .github/workflows/release.yml:59]: These are good questions.

**Review by alecjacobs5401 — APPROVED** [review]

**Comment by jmeridth** [comment]
@alecjacobs5401 when can we see a new release? Thank you so much for getting this change over the finish line. Cheers.

**Comment by alecjacobs5401** [comment]
@jmeridth We're finishing up some internal testing and potentially getting some other things cleaned up. I'll defer to @mckern on a release for all this

**Comment by mckern** [comment]
@alecjacobs5401 @jmeridth I need to write release notes and update the README. That's the blocker right now.

**Comment by jmeridth** [comment]
@mckern saw the release happened last week.  Thank you.

**Pull request merged**